File: shared/clang-wrapper/src/main/java/io/ygdrasil/koreos/clang/ClangException.java
// SPDX-License-Identifier: MIT
package io.ygdrasil.koreos.clang;

/**
 * Base exception for all Clang wrapper errors.
 */
public class ClangException extends RuntimeException {
    public ClangException(String message) {
        super(message);
    }

    public ClangException(String message, Throwable cause) {
        super(message, cause);
    }
}

File: shared/clang-wrapper/src/main/kotlin/io/ygdrasil/koreos/clang/ClangException.kt
// SPDX-License-Identifier: MIT
package io.ygdrasil.koreos.clang

/**
 * Base exception for all Clang wrapper errors.
 */
open class ClangException @JvmOverloads constructor(
    message: String,
    cause: Throwable? = null
) : RuntimeException(message, cause)
